evolver/hardware/standard/temperature.py

```diff
@@ -54,8 +60,16 @@ def read(self):
         response = self._do_serial()
         for vial, raw in enumerate(response.data):
             if vial in self.vials:
-                # calibration should happen here to populate temperature field from raw
-                self.outputs[vial] = self.Output(vial=vial, raw=int(raw))
+                # Calibrate raw data to temperature.
+                raw = int(raw)
+                temperature = None
+                if self.calibrator and self.calibrator.output_transformer:
+                    if isinstance(self.calibrator.output_transformer, dict):
+                        temperature = self.calibrator.output_transformer[vial].convert_to(raw)
+                    else:
+                        temperature = self.calibrator.output_transformer.convert_to(raw)
+
```

Comment:
Sorry @jamienoss , I should have just put it here. I mean that maybe we don't need to bother with the non-vial case, the procedure really should be producing one per vial even if there is only one vial

Reply:
Cool, thanks. I re-read your comment and realised what you were referring to.

I thought it was more useful and extensible to allow for both. I've also factored this out to the hardware interface as:
```python
    def _transform(self, transformer, func, x, vial=None):
        if self.calibrator and (_transformer := getattr(self.calibrator, transformer, None)):
            if isinstance(_transformer, dict):
                y = getattr(_transformer[vial], func)(x)
            else:
                y = getattr(_transformer, func)(x)
        else:
            y = x

        return y
```

Reply:
👍 ah, ok this will simplify while also being flexible